lib/modules/ogr/postgres.rb

```diff
@@ -35,4 +40,41 @@ def self.ogr_command template, context
     compiled_template = ERB.new(template).result context
     compiled_template.squish
   end
+
+  # The filename convention should be as follows:
+  #
+  # WDPA_MmmYYYY_Public if full WDPA (and only WDPA areas) download
+  # WDOECM_MmmYYYY_Public if full WDOECM (and only WDOECM areas) download
+  # WDPA_WDOECM_MmmYYYY_Public in all other scenarios.
+  #
+  # The GDB features naming convention follows the above with regards WDPA and WDOECM
+  # but with a minor amendment described as follows:
+  #
+  # WDPA_WDOECM_poly_MmmYYYY if polygon feature
+  # WDPA_WDOECM_point_MmmYYY if point feature
+  #
+  # So, given the original download filename,
+  # the WDPA_WDOECM bit depends on what the areas are about, e.g. only WDPA,
+  # only WDOECM or both. The 'Public' bit is removed and the date
+  # and the geometry type are swapped.
+  FEATURE_TYPES = {
+    'polygon' => 'poly',
+    'point' => 'point',
+    'source' => 'source'
+  }.freeze
+  def self.get_feature_name(filename, geom_type)
+    # Remove any possible folder structure from the path and also the extension
+    filename = filename.split('/').last[0..-5]
+    # Split the filename. E.g. "WDPA_MmmYYY_Public" => ['WDPA', 'MmmYYY', 'Public']
+    attrs = filename.split('_')
+    # Given the original filename should also contains 'polygons' or 'points' at the end,
+    # we remove this bit.
+    attrs.pop if %w(polygons points).include?(attrs[-1])
+    # Replace 'Public' with the geometry type formatted correctly, e.g. => ['WDPA', 'MmmYYY', 'poly']
+    attrs[-1] = FEATURE_TYPES[geom_type]
+    # Swap date and geometry type, e.g. => ['WDPA', 'poly', 'MmmYYY']
+    attrs[-1], attrs[-2] = attrs[-2], attrs[-1]
+    # Join and get the new filename, e.g. => 'WDPA_poly_MmmYYY'
+    attrs.join('_')
```

Comment:
Great explanation about the new naming conventions

Reply:
Just had a look though at Canada's GDB downloads as an example, and using Heather's comment in the Codebase ticket as context: 'The ISO3 code or WDPA ID can be added as a suffix where appropriate.' Shouldn't this be present at the end of the table name after 'Public'?

Reply:
Ah well spotted. 'Public' should have gone too, so what I think it happens is that for downloads with an identifier at the end, it removes the identifier instead of 'Public'. I'll change that.
Thanks!